From 107b5defb832064ae0dbec1ad9fb49e5b8234a1f Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Tue, 12 Mar 2013 09:01:23 +0000
Subject: Compressed files have a minmum size of 10240 bytes

libarchive: disable the padding of the last block

[bug #695672]
---
diff --git a/src/fr-archive-libarchive.c b/src/fr-archive-libarchive.c
index a48128d..462232c 100644
--- a/src/fr-archive-libarchive.c
+++ b/src/fr-archive-libarchive.c
@@ -1247,6 +1247,7 @@ save_archive_thread (GSimpleAsyncResult *result,
 	save_data->b = b = archive_write_new ();
 	_archive_write_set_format_from_context (b, save_data);
 	archive_write_open (b, save_data, save_data_open, save_data_write, save_data_close);
+	archive_write_set_bytes_in_last_block (b, 1);
 
 	a = archive_read_new ();
 	archive_read_support_filter_all (a);
--
cgit v0.9.1

