diff -Nur evince-3.6.1/libview/ev-annotation-window.c evince-3.6.1.new/libview/ev-annotation-window.c
--- evince-3.6.1/libview/ev-annotation-window.c	2012-06-24 16:30:36.000000000 +0200
+++ evince-3.6.1.new/libview/ev-annotation-window.c	2013-01-25 15:11:08.734467181 +0100
@@ -145,7 +145,6 @@
 
         properties = gtk_style_properties_new ();
         gtk_style_properties_set (properties, 0,
-                                  "color", &rgba,
                                   "background-color", &rgba,
                                   NULL);
 
