Index: gtk+3.0-3.8.4/gtk/gtkmodelmenu.c
===================================================================
--- gtk+3.0-3.8.4.orig/gtk/gtkmodelmenu.c	2013-11-21 19:43:24.000000000 +0000
+++ gtk+3.0-3.8.4/gtk/gtkmodelmenu.c	2013-11-21 21:05:35.917131771 +0000
@@ -343,6 +343,7 @@
   while (children)
     {
       gtk_container_remove (GTK_CONTAINER (binding->shell), children->data);
+      gtk_widget_destroy (children->data);
       children = g_list_delete_link (children, children);
     }
 
