Index: gtk+3.0-3.8.4/gtk/gtkappchooserwidget.c
===================================================================
--- gtk+3.0-3.8.4.orig/gtk/gtkappchooserwidget.c	2013-08-30 12:08:29.000000000 +0100
+++ gtk+3.0-3.8.4/gtk/gtkappchooserwidget.c	2013-11-21 20:52:17.753140994 +0000
@@ -536,9 +536,6 @@
           !g_app_info_supports_files (app))
         continue;
 
-      if (!g_app_info_should_show (app))
-        continue;
-
       if (g_list_find_custom (exclude_apps, app,
                               (GCompareFunc) compare_apps_func))
         continue;
