Description: Add Launchpad Integration
Author: ?

Index: gcalctool-6.0.1/configure.ac
===================================================================
--- gcalctool-6.0.1.orig/configure.ac	2011-04-21 18:11:03.000000000 +1000
+++ gcalctool-6.0.1/configure.ac	2011-04-21 18:13:54.337735190 +1000
@@ -45,6 +45,7 @@
     gio-2.0 >= $GIO_REQUIRED
     libxml-2.0
     gmodule-export-2.0
+    launchpad-integration-3.0
 ])
 
 PKG_CHECK_MODULES(GCALCCMD, [
Index: gcalctool-6.0.1/src/math-window.c
===================================================================
--- gcalctool-6.0.1.orig/src/math-window.c	2011-04-20 11:19:26.000000000 +1000
+++ gcalctool-6.0.1/src/math-window.c	2011-04-21 18:13:54.337735190 +1000
@@ -12,6 +12,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
+#include <launchpad-integration.h>
 
 #include "math-window.h"
 #include "math-preferences.h"
@@ -397,6 +398,7 @@
     menu = add_menu(window->priv->menu_bar, HELP_MENU_LABEL);
     menu_item = add_menu_item(menu, gtk_menu_item_new_with_mnemonic(HELP_CONTENTS_LABEL), G_CALLBACK(help_cb), window);
     gtk_widget_add_accelerator(menu_item, "activate", accel_group, GDK_KEY_F1, 0, GTK_ACCEL_VISIBLE);
+    launchpad_integration_add_items (menu, 1, TRUE, TRUE);
     add_menu_item(menu, gtk_image_menu_item_new_from_stock(GTK_STOCK_ABOUT, accel_group), G_CALLBACK(about_cb), window);
 }
 
