Description: Add Launchpad integration
Author: Robert Ancell <robert.ancell@canonical.com>

Index: gnome-disk-utility-3.0.0/configure.ac
===================================================================
--- gnome-disk-utility-3.0.0.orig/configure.ac	2011-04-04 17:51:53.000000000 +0200
+++ gnome-disk-utility-3.0.0/configure.ac	2011-05-16 15:44:08.126773235 +0200
@@ -151,6 +151,7 @@
 PKG_CHECK_MODULES(X11, [x11])
 PKG_CHECK_MODULES(LIBATASMART, [libatasmart >= 0.14])
 PKG_CHECK_MODULES(AVAHI_UI, [avahi-ui-gtk3 >= $AVAHI_UI_REQUIRED])
+PKG_CHECK_MODULES(LAUNCHPAD_INTEGRATION, [launchpad-integration-3.0])
 
 # *************
 # Remote Access
Index: gnome-disk-utility-3.0.0/src/palimpsest/Makefile.am
===================================================================
--- gnome-disk-utility-3.0.0.orig/src/palimpsest/Makefile.am	2011-04-04 17:48:16.000000000 +0200
+++ gnome-disk-utility-3.0.0/src/palimpsest/Makefile.am	2011-05-16 15:44:50.866773241 +0200
@@ -32,6 +32,7 @@
 	$(DBUS_GLIB_CFLAGS)				\
 	$(GNOME_KEYRING_CFLAGS)				\
 	$(GTK3_CFLAGS)					\
+	$(LAUNCHPAD_INTEGRATION_CFLAGS)			\
 	$(UNIQUE_CFLAGS)				\
 	$(X11_CFLAGS)					\
 	$(WARN_CFLAGS)					\
@@ -47,6 +48,7 @@
 	$(DBUS_GLIB_LIBS)				\
 	$(GNOME_KEYRING_LIBS)				\
 	$(GTK3_LIBS)					\
+	$(LAUNCHPAD_INTEGRATION_LIBS)			\
 	$(UNIQUE_LIBS)					\
 	$(X11_LIBS)					\
 	$(INTLLIBS)					\
Index: gnome-disk-utility-3.0.0/src/palimpsest/gdu-shell.c
===================================================================
--- gnome-disk-utility-3.0.0.orig/src/palimpsest/gdu-shell.c	2011-04-04 17:48:16.000000000 +0200
+++ gnome-disk-utility-3.0.0/src/palimpsest/gdu-shell.c	2011-05-16 15:43:53.076773233 +0200
@@ -29,6 +29,7 @@
 #include <glib-object.h>
 #include <string.h>
 #include <glib/gi18n.h>
+#include <launchpad-integration.h>
 
 #include <gdu/gdu.h>
 #include <gdu-gtk/gdu-gtk.h>
@@ -829,6 +830,7 @@
 #if 0
         "      <menuitem action='contents'/>"
 #endif
+        "      <placeholder name='LaunchpadItems'/>"
         "      <menuitem action='about'/>"
         "    </menu>"
         "  </menubar>"
@@ -870,6 +872,9 @@
                 gtk_main_quit ();
         }
 
+        /* Add launchpad hooks */
+        launchpad_integration_add_ui (ui_manager, "/menubar/help/LaunchpadItems");
+
         return ui_manager;
 }
 
